# 复利终值函数

计算现在金额未来的价值

```numpy_financial.fv(rate, nper, pmt, pv, when='end')```
rate为利率

nper为期数

pmt为每期固定支付款

pv为终值

when为期初还是期末付，1表示期初计数，0表示期末计数，默认为0

备注一下，负数是付出，正数是收到。

## 示例
假设，有一个项目，年利率为5%，每年年初现金流入10万元，5年后有多少钱？

In [1]:
import numpy as np
import numpy_financial as npf

r1=npf.fv(rate=0.05, nper=5-1, pmt=-100000, pv=-100000,when=1)

print("终值为：%.2f元" % r1)

终值为：574113.75元


如果为年终现金流入，只需要将FV函数的参数when改为0或删除。

In [2]:
r1=npf.fv(rate=0.05, nper=5-1, pmt=-100000, pv=-100000)
print("终值为：%.2f元" % r1)

终值为：552563.13元


年利率为5%，存100，然后每个月存100，10年后可以得到多少

In [3]:
npf.fv(0.05/12, 10*12, -100, -100)

15692.928894335748

# 现值PV函数

计算未来金额在现在的价值

```numpy_financial.pv(rate, nper, pmt,fv=0.0, when='end') #计算现值```
参数与fv相同

## 示例
假设银行的年利率为5%，每月投入1万元，现在需要投入多少本金才可以在5年后得到150万元？

In [4]:
r1=npf.pv(rate=0.05/12, nper=5*12, pmt=-10000, fv=1500000)
print("现值为：%.2f元" % r1)

现值为：-638901.02元


亦即现在需要投入63.89万元，5年后就可以有150万。

# 计算净现值

```numpy_financial.npv(rate, values)#计算净现值```

这个是审计中经常要用到的计算，净现值。净现值主要算的是刨去货币贬值影响最后还能赚多少钱。

净现值>0表示项目实施后，除保证可实现预定的收益率外，尚可获得更高的收益。

净现值<0表示项目实施后，未能达到预定的收益率水平，而不能确定项目已亏损。

净现值=0表示项目实施后的投资收益率正好达到预期，而不是投资项目盈亏平衡。

参数rate为折现率

参数values为收入，每一期的现金流。

## 示例

假设A项目的折现率为5%，前期均需投入100万元，最终收益为110万元，其一至五年的收入分别是10、20、20、30、30万元。判断该项目是否值得投资

In [5]:
import numpy_financial as npf

C_A=[-100, 10, 20, 20, 30, 30]

npv_A=npf.npv(0.05,C_A)

print("项目A的净现值：%.2f万元" % npv_A)

项目A的净现值：-6.87万元


结论是不值得投资

如果是两个相同的方案，收益相同，投入相同，但现金流入不同呢？这也可以计算。

假设有A、B两个项目，折现率均为5%，前期均需投入100万， 

A项目第一年至五年分别收入30、20、40、10、20万，共120万元，

而项目B第一至五年分别收入30、40、30、10、10万，也是120万元，

判断项目A和B哪个投资价值高？

In [6]:
import numpy_financial as npf
C_A=[-100, 30, 20, 40, 10, 20]
C_B=[-100, 30, 40, 30, 10, 10]
npv_A=npf.npv(0.05,C_A)
npv_B=npf.npv(0.05,C_B)
print("项目A的净现值：%.2f万元" % npv_A)
print("项目B的净现值：%.2f万元" % npv_B)

项目A的净现值：5.16万元
项目B的净现值：6.83万元


结论是项目B的收益大

# 计算内部收益率

```numpy_financial.irr(values)```

## 示例

还是以上面为例子，前期均需投入100万， A项目第一年至五年分别收入30、20、40、10、20万，共120万元，这个项目的内部收益率为多少呢？

In [7]:
import numpy_financial as npf
C_A=[-100, 10, 20, 20, 30, 30]
npv_A=npf.irr(C_A)
print("项目A的内部报酬率：%.2f" % npv_A)

项目A的内部报酬率：0.03


解读：我们每年能接受的货币贬值率为3%

# 计算贷款每期还款金额pmt

```numpy_financial.pmt(rate, nper, pv, fv=0, when="end")#计算每期应还的本金加利息```

## 示例

如果项目A的投入100万元不是自有资金，而是银行贷款，贷款利率为6%，分五年还清，那么每期应该要还多少钱呢？

import numpy as np
npv_A=np.pmt(0.06,5,1000000)
print("项目A每年应还贷：%.2f" % npv_A)

结论为每年还款23.73万元

计算月供：总价500万，30年还清， 年利率为5%

In [8]:
npf.pmt(0.05/12, 30*12, 5000000)

-26841.08115060699

结论：每月还款金额26841.08元

# 付款期数函数

```numpy_financial.nper(rate, pmt, pv, fv=0, when='end')```

参数rate为利率，pmt为每期支付金额，pv为现值。

如果这个项目我每年不能支付23万元，每年预计只能支付10万元，那么我应该向银行申请多少年还款比较合适呢？

In [9]:
import numpy_financial as npf
npv_A=npf.nper(0.06,-100000,1000000)
print("项目A计划贷款期数：%.2f" % npv_A)

项目A计划贷款期数：15.73


结论：可以向银行申请16年贷款